test/dkim.js

```diff
@@ -479,8 +478,31 @@ exports["Sign+verify tests"] = {
             }
             test.done();
         });
-    }
-}
+    },
+  "Warning on signing header missing": function(test) {
+    var mail = testMsg();
+    var dkimField = signMsg(mail, "node.ee", "dkim");
+    dkim.keyFromDNS = stubDNS;
+    dkim.DKIMVerify(dkimField + "\r\n" + mail, function(err, result) {
+      test.equal(err, null);
+      test.equal(result.sigs[0].warnings.length, 2);
+      test.ok(result.sigs[0].warnings[0].indexOf('is strongly recommended!'));
+      test.done();
+    });
+  },
+  "No warning if no header missing": function(test) {
```

Comment:
i was talking about this test. it's failing now as it'll have warnings for missing `date` in `h`.